testutil/testpolylog/event.go

```diff
@@ -0,0 +1,166 @@
+package testpolylog
+
+import (
+	"bytes"
+	"fmt"
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+
+	"github.com/pokt-network/poktroll/pkg/polylog"
+)
+
+type (
+	// NewLoggerAndOutputFn is called in the test helper to create a new logger
+	// configured with the given level and options. It returns the logger and
+	// the buffer to which the logger writes. It is useful for decoupling test
+	// helpers from a specific logger implementation.
+	NewLoggerAndOutputFn func(
+		*testing.T,
+		polylog.Level,
+		...polylog.LoggerOption,
+	) (polylog.Logger, *bytes.Buffer)
+
+	// NewEventWithLevelFn is called in the test helper to create a new event
+	// at the given level from the given logger. It is useful for decoupling
+	// test helpers from a specific logger implementation.
+	NewEventWithLevelFn func(
+		*testing.T,
+		polylog.Logger,
+		polylog.Level,
+	) polylog.Event
+)
+
+// FnMethodSpy is a mock which implements a #Fn() method that is intended to be
+// used in tests to assert that the function passed to polylog.Event#Func() is
+// called with the expected arg(s).
+type FnMethodSpy struct{ mock.Mock }
+
+// Fn is a mock method which can be asserted on via the mock.Mock API.
+// See: https://pkg.go.dev/github.com/stretchr/testify@v1.8.4/mock#Mock.
+func (m *FnMethodSpy) Fn(event polylog.Event) {
+	m.Called(event)
+}
+
+// EventMethodsTest is a test case for expressing and exercising polylog.Event
+// methods in a concise way.
+type EventMethodsTest struct {
+	Msg                    string
+	MsgFmt                 string
+	MsgFmtArgs             []any
+	Key                    string
+	Value                  any
+	EventMethodName        string
+	ExpectedOutputContains string
+}
+
+// RunEventMethodTests runs a set of tests for a given level.
+func RunEventMethodTests(
+	t *testing.T,
+	level polylog.Level,
+	tests []EventMethodsTest,
```

Comment:
Rename to `testCases` if we rename the above

Reply:
How strongly do you feel about  this? I recently switch my personal convention **from** `testCases` to `tests` (and have been doing so elsewhere in this codebase). I take your point that if the struct name includes the work "case" then that's clearly what we're talking about here. However,  I feel like it's unnecessarily verbose and I've since seen multiple examples of code in the wild (incl. cosmos-sdk) that uses the name `tests` conventionally for this purpose.

Reply:
Fair enough. Don't feel too strongly.

I feel more strongly about using `desc` instead of `name`, which we started doing in various places.

Reply:
:handshake: on `tests` and `desc`? :sweat_smile: